From 9b8feb254f6398477129bf5b1a97c74e5ee4f6d4 Mon Sep 17 00:00:00 2001
Message-Id: <9b8feb254f6398477129bf5b1a97c74e5ee4f6d4.1415496071.git.chang-joon.lee@intel.com>
In-Reply-To: <aa0c58fe046f9d00f3d381a75f6169b3f6d58c94.1415496071.git.chang-joon.lee@intel.com>
References: <aa0c58fe046f9d00f3d381a75f6169b3f6d58c94.1415496071.git.chang-joon.lee@intel.com>
From: Brad Volkin <bradley.d.volkin@intel.com>
Date: Wed, 21 May 2014 12:48:30 -0700
Subject: [PATCH 4/8] MUST_REBASE [VPG]: drm/i915: Change cmd parser version
 number

Since MUST_REBASE Android patches modify the behavior of upstream
command parsing, we should also modify the version number as the
intention is for userspace to rely on the version number to know
what rules the kernel enforces.

For: APDEV-870
Change-Id: Ia481fdbe21a4bf49abd4a95255b249b3c5034261
Signed-off-by: Brad Volkin <bradley.d.volkin@intel.com>
---
 drivers/gpu/drm/i915/i915_dma.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index 734a455..3c488f1 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -1023,6 +1023,8 @@ static int i915_getparam(struct drm_device *dev, void *data,
 		break;
 	case I915_PARAM_CMD_PARSER_VERSION:
 		value = i915_cmd_parser_get_version();
+		/* Adjust the parser version for Android */
+		value |= (value << 16);
 		break;
 	case I915_PARAM_EU_COUNT:
 		value = INTEL_INFO(dev)->eu_total;
-- 
1.7.9.5

